frontend/demo/component/button/button-disable-long-action.ts

```diff
@@ -0,0 +1,51 @@
+import 'Frontend/demo/init'; // hidden-full-source-line
+
+import { html, LitElement, customElement, internalProperty } from 'lit-element';
+import '@vaadin/vaadin-button/vaadin-button';
+import '@vaadin/vaadin-progress-bar/vaadin-progress-bar';
+import { applyTheme } from 'Frontend/generated/theme';
+
+@customElement('button-disable-long-action')
+export class Example extends LitElement {
+  constructor() {
+    super();
+    // Apply custom theme (only supported if your app uses one)
+    applyTheme(this.shadowRoot);
+  }
+
+  @internalProperty()
+  progressBarValue = 0;
+
+  render() {
+    return html`
+      <!-- tag::snippet[] -->
+      <vaadin-button @click=${this.performAction}>Perform Action</vaadin-button>
+      <vaadin-progress-bar
+        value="${this.progressBarValue}"
+        style="display: inline-block; width: 350px;"
+      ></vaadin-progress-bar>
+      <!-- end::snippet[] -->
+    `;
+  }
+
+  performAction() {
+    const button = this.shadowRoot?.querySelector('vaadin-button');
+    if (button) {
+      // disable the button
+      button.disabled = true;
```

Comment:
```suggestion
      button.disabled = true;  // <1>
```

Add a noticeable callout here, to make it easier to spot the relevant part of the code.

Reply:
Rather than manually toggling the button's disabled property, you should bind it to the state of your UI. Quickly looking at the logic, the button should be disabled when the progress isn't running. You could indicate non-started progress for example by having the `progressBarValue` undefined.

So the state binding could looks something like `<vaadin-button .disabled="${this.progressBarValue !== undefined}"`